app/command_autocomplete_test.go

```diff
@@ -609,3 +611,66 @@ func createJiraAutocompleteData() *model.AutocompleteData {
 
 	return jira
 }
+
+func TestDynamicListArgsForBuiltin(t *testing.T) {
+	th := Setup(t)
+	defer th.TearDown()
+
+	provider := &testProvider{}
+	RegisterCommandProvider(provider)
+
+	command := provider.GetCommand(th.App, nil)
+	emptyCmdArgs := &model.CommandArgs{}
+
+	t.Run("GetAutoCompleteListItems", func(t *testing.T) {
+		suggestions := th.App.getSuggestions(emptyCmdArgs, []*model.AutocompleteData{command.AutocompleteData}, "", "bogus --dynaArg ", model.SYSTEM_ADMIN_ROLE_ID)
+		assert.Len(t, suggestions, 3)
+		assert.Equal(t, "this is hint 1", suggestions[0].Hint)
+		assert.Equal(t, "this is hint 2", suggestions[1].Hint)
+		assert.Equal(t, "this is hint 3", suggestions[2].Hint)
+	})
+
+	t.Run("GetAutoCompleteListItems bad arg", func(t *testing.T) {
+		suggestions := th.App.getSuggestions(emptyCmdArgs, []*model.AutocompleteData{command.AutocompleteData}, "", "bogus --badArg ", model.SYSTEM_ADMIN_ROLE_ID)
+		assert.Empty(t, suggestions)
+	})
+}
+
+type testProvider struct {
```

Comment:
nit: we can use more descriptive naming. 
```suggestion
type testCommandProvider struct {
```

Reply:
Done.  I can't use your commit suggestion as the name needs to change in multiple places.